server/events/project_command_builder.go

```diff
@@ -288,6 +288,34 @@ func (p *DefaultProjectCommandBuilder) buildPlanAllCommands(ctx *CommandContext,
 			return nil, errors.Wrapf(err, "parsing %s", yaml.AtlantisYAMLFilename)
 		}
 		ctx.Log.Info("successfully parsed %s file", yaml.AtlantisYAMLFilename)
+
+		// Construct deletion plans if flag is enabled
+		if repoCfg.DeleteAllProjects {
```

Comment:
This is a good approach, but I think one way to minimize changes to the atlantis codebase is to leverage [Tags](https://github.com/lyft/atlantis/blob/cd71dab4ec6baf72d5f0e463f949807fb811872f/server/events/yaml/valid/repo_cfg.go#L110-L111)

I added this to allow us to pass in dynamic attributes for projects so in this case, we could pass in "destroy" or something as an attribute and that is plumbed through to the [ProjectCommandContext](https://github.com/lyft/atlantis/blob/cd71dab4ec6baf72d5f0e463f949807fb811872f/server/events/models/models.go#L412-L413).  You can then use that to set the args in the plan command runner.


Then in order to trigger this, in our atlantis.yaml generation code (in our internal repo), we could just set the when_modified dynamically to the manifest as the trigger.

Does that make sense? Thoughts on this approach?